analyzers/src/SonarAnalyzer.Common/Helpers/ParameterLoader.cs

```diff
@@ -39,84 +36,27 @@ internal static class ParameterLoader
          * - diffing the contents of the configuration file
          * - associating the file with a unique identifier for the build project
          */
-        internal static void SetParameterValues(ParametrizedDiagnosticAnalyzer parameteredAnalyzer,
-            AnalyzerOptions options)
+        internal static void SetParameterValues(ParametrizedDiagnosticAnalyzer parameteredAnalyzer, SonarLintXmlReader sonarLintReader)
         {
-            var sonarLintXml = options.SonarLintXml();
-            if (sonarLintXml == null)
-            {
-                return;
-            }
-
-            var parameters = ParseParameters(sonarLintXml);
-            if (parameters.IsEmpty)
```

Comment:
This sort of check should remain.  `sonarLintReader.ParametrizedRules.Any()`
Why? - There is no need to iterate through the whole set of `propertyParameterPairs` if there is no configured parameter value in the SonarLint.xml file.